### PR TITLE
Replace ioutil package with os and io package

### PR DESCRIPTION
**Pull request by tenzen-y**
**What this PR does / why we need it**:
I replaced ioutil package with os and io package since that package has been deprecated.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/router/main.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func callService(serviceUrl string, input []byte, headers http.Header) ([]byte,
 		return nil, err
 	}
 	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
+	body, err := io.ReadAll(resp.Body)
```
Comment (tenzen-y) [inline cmd/router/main.go:60]: > Deprecated: As of Go 1.16, this function simply calls io.ReadAll.

https://pkg.go.dev/io/ioutil#ReadAll

File: pkg/agent/downloader.go [hunk 2]
```diff
@@ -61,7 +60,7 @@ func (d *Downloader) DownloadModel(modelName string, modelSpec *v1alpha1.ModelSp
 			if err != nil {
 				return errors.Wrapf(createErr, "failed to encode model spec")
 			}
-			err = ioutil.WriteFile(successFile, encodedJson, 0644)
+			err = os.WriteFile(successFile, encodedJson, 0644)
```
Comment (tenzen-y) [inline pkg/agent/downloader.go:63]: > Deprecated: As of Go 1.16, this function simply calls os.WriteFile.

https://pkg.go.dev/io/ioutil#WriteFile

File: pkg/agent/downloader_test.go [hunk 3]
```diff
@@ -34,7 +33,7 @@ var _ = Describe("Downloader", func() {
 	var modelDir string
 	var downloader *Downloader
 	BeforeEach(func() {
-		dir, err := ioutil.TempDir("", "example")
+		dir, err := os.MkdirTemp("", "example")
```
Comment (tenzen-y) [inline pkg/agent/downloader_test.go:36]: > Deprecated: As of Go 1.17, this function simply calls os.MkdirTemp.

https://pkg.go.dev/io/ioutil#TempDir

File: pkg/agent/watcher.go [hunk 4]
```diff
@@ -62,7 +62,7 @@ type modelWrapper struct {
 }
 
 func (w *Watcher) syncModelConfig(modelConfigFile string, initializing bool) error {
-	file, err := ioutil.ReadFile(modelConfigFile)
+	file, err := os.ReadFile(modelConfigFile)
```
Comment (tenzen-y) [inline pkg/agent/watcher.go:65]: > Deprecated: As of Go 1.16, this function simply calls os.ReadFile.

https://pkg.go.dev/io/ioutil#ReadFile

File: pkg/logger/handler.go [hunk 5]
```diff
@@ -104,7 +104,7 @@ func (eh *LoggerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Proxy Request
-	r.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+	r.Body = io.NopCloser(bytes.NewBuffer(body))
```
Comment (tenzen-y) [inline pkg/logger/handler.go:107]: > Deprecated: As of Go 1.16, this function simply calls io.NopCloser.

https://pkg.go.dev/io/ioutil#NopCloser

**Comment by yuzisun** [comment]
Thanks @tenzen-y !!

/lgtm
/approve

**Pull request merged**